src/payment/strategies/paypal-commerce/paypal-commerce-alternative-methods-strategy.ts

```diff
@@ -0,0 +1,145 @@
+    PaypalCommerceCreditCardPaymentInitializeOptions,
+    PaypalCommerceInitializationData,
+    PaypalCommercePaymentInitializeOptions,
+    PaypalCommercePaymentProcessor, PaypalCommerceScriptParams,
+    PaypalCommerceSDKFunding } from './index';
+
+export default class PaypalCommerceAlternativeMethodsStrategy implements PaymentStrategy {
+    private _orderId?: string;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _paypalCommercePaymentProcessor: PaypalCommercePaymentProcessor
+    ) {}
+
+    async initialize({methodId, gatewayId, paypalcommerce}: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+
+        const multiOptionId = methodId;
+        let fundingKey: keyof PaypalCommerceSDKFunding;
+        if (gatewayId) {
+            methodId = gatewayId;
+        }
+
+        const { paymentMethods: { getPaymentMethodOrThrow }, cart: { getCartOrThrow } } = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId));
+        const { initializationData } = getPaymentMethodOrThrow(methodId);
+        const { buttonStyle } = initializationData;
+
+        if (initializationData.orderId) {
+            this._orderId = initializationData.orderId;
+
+            return this._store.getState();
+        }
+
+        if (!paypalcommerce) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.paypalcommerce" argument is not provided.');
+        }
+
+        if (!this._isPaypalCommerceOptionsPayments(paypalcommerce)) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.paypalcommerce" argument should contain "container", "onRenderButton", "submitForm".');
+        }
+
+        const { container, onRenderButton, submitForm } = paypalcommerce;
+        const { id: cartId, currency: { code: currencyCode } } = getCartOrThrow();
+
+        const paramsScript = this._getOptionsScript(initializationData, currencyCode);
+        const buttonParams: ButtonsOptions = { style: buttonStyle, onApprove: data => this._tokenizePayment(data, submitForm) };
+
+        await this._paypalCommercePaymentProcessor.initialize(paramsScript);
+        switch (multiOptionId) {
+            case 'Bancontact':
+                fundingKey = 'BANCONTACT';
+                break;
+            case 'giropay':
+                fundingKey = 'GIROPAY';
+                break;
+            case 'Przlewy24':
+                fundingKey = 'PRZLEWY24';
+                break;
+            default:
+                fundingKey = 'BANCONTACT';
+        }
+
+        this._paypalCommercePaymentProcessor.renderButtons(cartId, container, buttonParams, {
+            onRenderButton,
+            // eslint-disable-next-line @typescript-eslint/tslint/config
```

Comment:
Can we find a way to follow the lint rule rather than trying to bypass it?

Reply:
removed file as it is no longer needed